### PR TITLE
Add hasAuthorization to Query

### PR DESCRIPTION
**Pull request by mwizeman**
This will allow a search to find all elements that use a particular authorization

### CODE REVIEW

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java [hunk 1]
```diff
@@ -583,6 +585,40 @@ private QueryBuilder getFilterForHasExtendedDataFilter(HasExtendedDataFilter has
         return boolQuery;
     }
 
+    protected QueryBuilder getFilterForHasAuthorizationContainer(HasAuthorizationContainer hasAuthorization) {
+        PropertyNameVisibilitiesStore visibilitiesStore = getSearchIndex().getPropertyNameVisibilitiesStore();
+        Authorizations auths = getParameters().getAuthorizations();
+        Graph graph = getGraph();
+
+        Set<String> hashes = StreamSupport.stream(hasAuthorization.getAuthorizations().spliterator(), false)
```
Comment (joeferner) [inline elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java]: Use `StreamUtils.stream` (it's a Vertexium class)

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java [hunk 2]
```diff
@@ -583,6 +585,40 @@ private QueryBuilder getFilterForHasExtendedDataFilter(HasExtendedDataFilter has
...
+
+        List<QueryBuilder> filters = new ArrayList<>();
+        for (PropertyDefinition propertyDefinition : graph.getPropertyDefinitions()) {
+            String propertyName = propertyDefinition.getPropertyName();
+
+            Set<String> matchingPropertyHashes = visibilitiesStore.getHashes(graph, propertyName, auths).stream()
+                    .filter(hashes::contains)
+                    .collect(Collectors.toSet());
+            for (String fieldName : getSearchIndex().addHashesToPropertyName(propertyName, matchingPropertyHashes)) {
+                filters.add(QueryBuilders.existsQuery(fieldName.replace(".", FIELDNAME_DOT_REPLACEMENT)));
```
Comment (joeferner) [inline elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java]: Can we make this `fieldName.replace....` a static method somewhere.

File: elasticsearch-singledocument/src/main/java/org/vertexium/elasticsearch/ElasticSearchSingleDocumentSearchQueryBase.java [hunk 3]
```diff
@@ -540,6 +543,40 @@ private FilterBuilder getFilterForHasExtendedDataFilter(HasExtendedDataFilter ha
         return FilterBuilders.andFilter(filters.toArray(new FilterBuilder[filters.size()]));
     }
 
+    protected FilterBuilder getFilterForHasAuthorizationContainer(HasAuthorizationContainer hasAuthorization) {
+        PropertyNameVisibilitiesStore visibilitiesStore = getSearchIndex().getPropertyNameVisibilitiesStore();
+        Authorizations auths = getParameters().getAuthorizations();
+        Graph graph = getGraph();
+
+        Set<String> hashes = StreamSupport.stream(hasAuthorization.getAuthorizations().spliterator(), false)
```
Comment (joeferner) [inline elasticsearch-singledocument/src/main/java/org/vertexium/elasticsearch/ElasticSearchSingleDocumentSearchQueryBase.java]: Same here... use `StreamUtils.stream`

**Comment by mwizeman** [comment]
@joeferner Updated to incorporate your comments with 10379557d617adde11fc801b787d2473b031c26b

**Pull request merged**